base_layer/wallet/src/base_node_service/monitor.rs

```diff
@@ -198,24 +216,27 @@ where
         Ok(())
     }
 
-    async fn update_state(&self, new_state: BaseNodeState) {
+    // returns true if a new block, otherwise false
+    async fn update_state(&self, new_state: BaseNodeState) -> bool {
```

Comment:
I think we should monitor changes to the base `node_id` here as well (not all the rest of  `BaseNodeState`) - if it changed then this function should return true as well. That way, if a base node is swapped, we can trigger `validate_outputs` to run upon that change,

Reply:
Resolved with comment below